File: CodeSchool.Domain/UserSubTask.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CodeSchool.Domain
{
    public enum UserSubTaskAnswerScore
    {
        DontKnow = 0,
        HardToRemember,
        KnowIt
    }

    public class UserSubTask
    {
        public UserSubTask()
        {
            CodeSnapshots = new List<CodeSnapshot>();
        }

        [Required]
        public int Id { get; set; }

        public int SubTaskId { get; set; }

        public int UserTaskHeadId { get; set; }

        public Guid UserId { get; set; }

        public bool IsPassed { get; set; }

        public string Code { get; set; }

        public DateTime? CreatedDt { get; set; }
        public DateTime? UpdatedDt { get; set; }

        // Maybe will be removed in future
        public int? SelectedAnswerOptionId { get; set; }

        public UserSubTaskAnswerScore? Score { get; set; }
        
        // Virtual properties
        public virtual User User { get; set; }
        public virtual UserTaskHead UserTaskHead { get; set; }
        public virtual SubTask SubTask { get; set; }
        public virtual ICollection<CodeSnapshot> CodeSnapshots { get; set; }
    }
}

File: CodeSchool.Web/ClientApp/app/models/usersubtask.ts
import { Mapper } from "../utils/helpers";
import { SubTaskViewModel } from "./sub-task";

export enum UserSubTaskAnswerScore {
    DontKnow = 0,
    HardToRemember,
    KnowIt
}

export class UserSubTaskModel {
    constructor(model?: UserSubTaskModel) {
        if (model) {
            Mapper.map(model, this);
        }
    }

    id: number;
    subTaskId: number;
    userTaskHeadId: number;
    selectedAnswerOptionId?: number;
    score?: UserSubTaskAnswerScore;
    userId: string;
    isPassed: boolean;
    code: string;
    createdDt: any;
    subTaskTitle: string;
    subTaskOrder: number;
    timeLimit: number;
    subTask: SubTaskViewModel;
}